### PR TITLE
Add docker action

### PR DESCRIPTION
**Pull request by Sammers21**
Will be ready for review as soon as #43 will be merged

The second step of #41

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#44](https://github.com/artipie/artipie/issues/44) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/ce82bd45-2fdd-4ac7-aa81-147cb92f735d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by fanifieiev** [comment]
@Sammers21 is it ready for review?
#43 has been merged

**Comment by Sammers21** [comment]
@fanifieiev, yes, it is ready now.

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: .github/workflows/docker-release.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Publish Docker
+on:
+  create:
+    tags:
+      - .*
+  push:
+    branches:
+      - .*
```
Comment (fanifieiev) [inline .github/workflows/docker-release.yml]: @Sammers21 Isn't is better to specify `master` branch only here?

File: .github/workflows/docker-release.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Publish to Registry
```
Comment (fanifieiev) [inline .github/workflows/docker-release.yml]: @Sammers21 What about `Publish Image to Docker Registry` ?

File: .github/workflows/docker-release.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Publish Docker
```
Comment (fanifieiev) [inline .github/workflows/docker-release.yml]: @Sammers21 What about `Publish Docker Image` ?

File: .github/workflows/docker-release.yml (hunk 1 above)
Comment (Sammers21) [reply .github/workflows/docker-release.yml]: Pull request branches does not count, that is why i think it is ok

File: .github/workflows/docker-release.yml (hunk 3 above)
Comment (Sammers21) [reply .github/workflows/docker-release.yml]: I would change it to `Docker Release`, just for consistency with file name

File: .github/workflows/docker-release.yml (hunk 1 above)
Comment (fanifieiev) [reply .github/workflows/docker-release.yml]: @Sammers21 It actually should fine, the only edge case would be if someone creates extra branch directly.
Ok, let it be.

File: .github/workflows/docker-release.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  push:
+    branches:
+      - .*
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Publish Image to Docker Registry
+        uses: elgohr/Publish-Docker-Github-Action@master
```
Comment (fanifieiev) [inline .github/workflows/docker-release.yml]: @Sammers21 By the way, how safe /stable is it to use someone's personal repo?
Say, what if that guy removes his repository?

Comment (Sammers21) [reply .github/workflows/docker-release.yml]: Stable enough. If he would, we can use the fork:https://github.com/Sammers21/Publish-Docker-Github-Action

Comment (fanifieiev) [reply .github/workflows/docker-release.yml]: Ok

**Comment by fanifieiev** [comment]
@g4s8 please merge

**Comment by 0crat** [comment]
Job [#44](https://github.com/artipie/artipie/issues/44) is not in the agenda of @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), can't retrieve data and time of add

**Comment by 0crat** [comment]
The project doesn't have enough funds, can't make a payment

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Comment by 0crat** [comment]
Code review was too long (11 days), architects (@g4s8) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Pull request merged**